### DISCUSSION TITLE
Is it possible to use the result of one testsuite into another testsuite?

### DISCUSSION THREAD

**Discussion by fabriziomoscon** [ovh/venom#510] [Q&A] [answered]
I have a generic testsuite step for m2m authentication, which would be great to share among all test suites.

This is the relevant code for the shared m2m test.

```
name: M2M

testcases:

  - name: m2m
    steps:
      -
        type: http
        info:
          - Simulates login via m2m
        [...]
        vars:
          token:
            from: result.bodyjson.data.servicetoken.token
            regex: .*
```

I would like to use it into another testsuite:

```
name: feature A

testcases:
 - name: User flow 1
    steps:
      -
        type: http
        info:
          - Simulates chat authentication with bearer token.
        [...]
        headers:
          "Authorization": "Bearer {{.M2M.m2m.serviceToken}}"
        [...]
```

Is it possible?

**Comment by nqb** [comment, marked as answer]
See https://github.com/ovh/venom/issues/241

For this use case, on your side we use a custom executor but we call it each time, see:
- custom executor: https://github.com/inverse-inc/packetfence/blob/f8b9c041d78163f9307c37e424e815050330b552/t/venom/lib/get_login_token.yml
- usage: https://github.com/inverse-inc/packetfence/blob/f8b9c041d78163f9307c37e424e815050330b552/t/venom/test_suites/global_config/00_create_roles.yml#L3-L23

**Reply by fabriziomoscon** [reply]
Thank you.
Do I need to implement golang code for `get_login_token` ? Or pass a special parameter to venom binary?

I received this error:

```
executor "get_login_token" is not implemented
```

**Reply by fabriziomoscon** [reply]
`--lib-dir` ?

**Reply by nqb** [reply]
I think it should work. On your side, we use `.venomrc`: https://github.com/inverse-inc/packetfence/blob/b60881999323df26be3f4f5151ca163954b6b056/t/venom/.venomrc#L7
